python/pyarrow/tests/test_pandas.py

```diff
@@ -4472,6 +4472,21 @@ def test_timestamp_as_object_non_nanosecond(resolution, tz, dt):
         assert result[0] == expected
 
 
+def test_timestamp_as_object_pytz_offset():
+    # ARROW-16547 to_pandas with timestamp_as_object=True and FixedOffset
+    pytz = pytest.importorskip("pytz")
+    import datetime
+    timezone = pytz.FixedOffset(120)
+    dt = timezone.localize(datetime.datetime(2022, 5, 12, 16, 57))
+
+    table = pa.table({"timestamp_col": pa.array([dt])})
+
+    expected = table.to_pandas()
```

Comment:
Instead of relying on the normal to_pandas conversion (and thus relying on the equality comparison between an actual pandas timestamp and datetime object), I would maybe check the result manually (you know that `dt` should roundtrip?)